test/test_pipeline_extractive_qa.py

```diff
@@ -98,3 +101,48 @@ def test_extractive_qa_answers_with_translator(
     assert (
         prediction["answers"][0].context == "My name is Carla and I live in Berlin"
     )
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("retriever_with_docs", ["elasticsearch", "dpr", "embedding"], indirect=True)
+@pytest.mark.parametrize("document_store_with_docs", ["elasticsearch"], indirect=True)
+@pytest.mark.parametrize("reader", ["farm"], indirect=True)
+def test_extractive_qa_es_authentication(reader, retriever_with_docs, document_store_with_docs: ElasticsearchDocumentStore):
+    if isinstance(retriever_with_docs, (DensePassageRetriever, EmbeddingRetriever)):
+        document_store_with_docs.update_embeddings(retriever=retriever_with_docs)
+    mock_client = Mock(wraps=document_store_with_docs.client)
+    document_store_with_docs.client = mock_client
+    auth_headers = {'Authorization': 'Basic YWRtaW46cm9vdA=='}
+    pipeline = ExtractiveQAPipeline(reader=reader, retriever=retriever_with_docs)
```

Comment:
Wondering if we can lower the memory footprint / runtime of this test further :thinking: 
Do we gain anything by having the reader in here? Wouldn't a `DocumentSearchPipeline` with a plain retriever be enough? Do we really want / need three retrievers here?

Reply:
Yep, we can do that... Just had the most common path in mind. But I totally agree with you, that memory footprint / runtime is currently a bigger pain. I chose the three such that the test ensures all retrievers compatible with ElasticsearchDocumentStore and actually searching with es propagate the headers correctly.

Reply:
done